version: '3'
services:
  db:
    image: "postgres:12.2"
    environment:
      POSTGRES_HOST_AUTH_METHOD: "trust"
    volumes:
      - tamato-data-volume:/var/lib/postgresql/data
    ports:
      - "5431:5432"

  cache-redis:
    image: redis
    restart: always
    ports:
      - "6379:6379"
    volumes:
      - .:/redis.conf

  celery-redis:
    image: redis
    restart: always
    ports:
      - "6378:6379"
    volumes:
      - .:/redis.conf

  s3:
    image: minio/minio
    restart: always
    ports:
      - "9003:9003"
    volumes:
      - tamato-s3-volume:/data
    env_file: .env
    command: server --quiet --address 0.0.0.0:9003 /data

  tamato:
    build: .
    image: tamato
    ports:
      - "8000:8000"
    depends_on:
      - db
      - s3
      - cache-redis
      - celery-redis
    links:
      - db
      - s3
      - cache-redis
      - celery-redis
    environment:
      DATABASE_URL: "postgres://postgres@db/postgres"
      DJANGO_SETTINGS_MODULE: "settings.dev"
      ENABLE_DJANGO_DEBUG_TOOLBAR: "False"
      PYTHONUNBUFFERED: "1"
      STARTING_TRANSACTION: "0"
      NURSERY_CACHE_ENGINE: "importer.cache.redis.RedisCacheEngine"
      CACHE_URL: "redis://cache-redis:6378/1"
      CELERY_BROKER_URL: "redis://celery-redis:6378/1"
      SSO_ENABLED: "false"
    command: python manage.py runserver 0.0.0.0:8000

volumes:
  tamato-data-volume:
  tamato-s3-volume:
